### PR TITLE
refactor the return and create one liner

### PR DESCRIPTION
**Pull request by nixbytes**
Hi, 

First of all, great project. I wanted to help out a little and figure I refactor of code on the client. Let me know if there is anything I need to do or change. so far the test 406 passed, 4 skipped, 1 xfailed in 15.21s

### CODE REVIEW

File: httpie/client.py [hunk 1]
```diff
@@ -254,13 +254,7 @@ def make_request_kwargs(
...
     auto_json = data and not args.form
     if (args.json or auto_json) and isinstance(data, dict):
-        if data:
-            data = json.dumps(data)
-        else:
-            # We need to set data to an empty string to prevent requests
-            # from assigning an empty list to `response.request.data`.
-            data = ''
-
+        data = json.dumps(data) if data else ''
```
Comment (BoboTiG) [inline httpie/client.py:257]: The comment is important to keep. Could you revert that change?

**Review by BoboTiG — CHANGES_REQUESTED** [review]
Thank you, I like that kind of clean-up.

**Comment by nixbytes** [comment]
No problem, I just want to get started on contribution to open source. :)

File: httpie/client.py (hunk 1 above)
Comment (nixbytes) [reply httpie/client.py:257]: sure I will update it today :)

**Comment by nixbytes** [comment]
added the missing comment

File: httpie/client.py [hunk 2]
```diff
@@ -320,5 +314,5 @@ def ensure_path_as_is(orig_url: str, prepped_url: str) -> str:
         **parsed_prepped._asdict(),
         'path': parsed_orig.path,
     }
-    final_url = urlunparse(tuple(final_dict.values()))
-    return final_url
+
```
Comment (BoboTiG) [inline httpie/client.py:317]:
```suggestion
```

File: httpie/client.py (hunk 1 above)
Comment (BoboTiG) [reply httpie/client.py:259]: I would prefer to not change the code: the old version is more clear. Do you mind not altering that part in tha PR?

**Review by BoboTiG — COMMENTED** [review]
And thanks for your contribution, that's a nice start :)

**Comment by nixbytes** [comment]
sure, I added the code back and did some refactor of the returns. let me know if this works for you

**Comment by nixbytes** [comment]
sorry had to revert the single quotes because of black, which I use to clean up code but didn't mean to change thing dramatically

File: httpie/client.py [hunk 3]
```diff
@@ -260,7 +258,6 @@ def make_request_kwargs(
             # We need to set data to an empty string to prevent requests
             # from assigning an empty list to `response.request.data`.
             data = ''
-
     # Finalize headers.
```
Comment (BoboTiG) [inline httpie/client.py:261]:
```suggestion

    # Finalize headers.
```

**Review by BoboTiG — APPROVED** [review]

File: httpie/client.py [hunk 4]
```diff
@@ -320,5 +316,5 @@ def ensure_path_as_is(orig_url: str, prepped_url: str) -> str:
         **parsed_prepped._asdict(),
         'path': parsed_orig.path,
     }
-    final_url = urlunparse(tuple(final_dict.values()))
-    return final_url
+
```
Comment (BoboTiG) [inline httpie/client.py:319]:
```suggestion
```

**Comment by BoboTiG** [comment]
@nixbytes :champagne: :)

**Comment by nixbytes** [comment]
let me know if anything else I can do to help.

**Pull request merged**